### PR TITLE
Fix failing tests

### PR DESCRIPTION
**Pull request by psiroky**
@mariofusco this should fix the test failure we have been seeing in CI (at least it does for me when running locally). We will have to run multiple builds in CI to be sure though.

### CODE REVIEW

**Comment by psiroky** [comment]
jenkins, retest this

**Comment by psiroky** [comment]
Looks like the current fix is not enough, some tests still failing.

**Comment by psiroky** [comment]
I've executed the build 20 times in a row on one of the CI machines and all the runs were successful. This does not mean there aren't any other tests influencing each other, but it is definitely a step forward.

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java [hunk 1]
```diff
@@ -190,9 +191,10 @@ public void run() {
             kieSession.setGlobal("list", list);
             kieSession.fireAllRules();
             kieSession.dispose();
-            // There can be 2 items in the list if an updateToVersion is triggered during a fireAllRules
-            // in that case it may fire with both the old and the new rule
-            assertTrue( list.size() >= 1 && list.size() <= 2);
+            // There can be multiple items in the list if an updateToVersion is triggered during a fireAllRules
+            // (updateToVersion can be called multiple times during fireAllRules, especially on slower machines)
+            // in that case it may fire with the old rule and multiple new ones
+            Assertions.assertThat(list.size()).isGreaterThanOrEqualTo(1);
```
Comment (triceo) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java]: Unless I'm mistaken, you can `assertThat(list)` and then use something like `.isNotEmpty()`.

Comment (psiroky) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java]: I maybe be missing something (I am not much familiar with AssertJ), but wouldn't that fail for lists with size greater than 1?

Comment (triceo) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java]: The original code is asserting that the list's size `isGreaterThanOrEqualTo(1)`. In other words - if there is at least one element in the list. Changing this to what I propose is just a more descriptive way of saying the same.

Comment (psiroky) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java]: `isNotEmpty()` works, I was originally looking at `hasSize()` which I though was in the first comment.... Anyway, thanks I will fix it.

Comment (triceo) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java]: Ah, my apologies - I fixed my original comment 5 seconds after I posted it, I had no idea someone was able to read it in the meantime. :-)

Comment (psiroky) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieContainerTest.java]: Looks like some caching issue as I read the comment like 15 mins after it was created (at least that what github was showing). Not important, though. Should be now fixed.

**Comment by etirelli** [comment]
Looks good @psiroky . +1

**Comment by psiroky** [comment]
Merged into master.

**Pull request closed without merging**